File: BestTimeToBuyAndSellStock.js
var maxProfit = function(prices) {
    var maxProfit = 0;
    var currProfit = 0;
    var start = prices[0];
    
    for (var i = 0; i < prices.length; i++) {
        if (prices[i] < start) {
            start = prices[i];
        } else {
            currProfit = prices[i] - start;
            maxProfit = Math.max(maxProfit, currProfit);
        }
        
    }
    
    return maxProfit;
};

File: BestTimeToBuyAndSellStock.py
class BestTimeToBuyAndSellStocks(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
            
        buy = prices[0]
        maxProfit = 0
        for i in range (len(prices)):
            if prices[i] < buy:
                buy = prices[i]
            else:
                maxProfit = max(maxProfit, prices[i] - buy)

        return maxProfit